### PR TITLE
Upgrade to circe-0.12.0-M4, drop scala-2.11

### PR DESCRIPTION
**Pull request by rossabaker**
Discussion at #2660.

### CODE REVIEW

File: core/src/main/scala/org/http4s/QValue.scala [hunk 1]
```diff
@@ -129,7 +129,7 @@ object QValue {
   @deprecated("""use qValue"" string interpolation instead""", "0.20")
   def q(d: Double): QValue = macro Macros.qValueLiteral
 
-  implicit val http4sOrderForQValue: Order[QValue] = Order.fromOrdering[QValue]
+  implicit lazy val http4sOrderForQValue: Order[QValue] = Order.fromOrdering[QValue]
```
Comment (rossabaker) [inline core/src/main/scala/org/http4s/QValue.scala]: It's not immediately clear to me why this became necessary.  Without it, we get:

```scala
[error] cannot create an instance for class org.http4s.testing.ResponseGeneratorSpec
[error]   caused by java.lang.ExceptionInInitializerError
[error]   caused by scala.UninitializedFieldError: Uninitialized field: /home/travis/build/http4s/http4s/core/src/main/scala/org/http4s/QValue.scala: 132
```

Comment (rossabaker) [reply core/src/main/scala/org/http4s/QValue.scala]: Okay, this is now causing a stack overflow error.  I'm reaching the stupid hours... what's suddenly wrong with this?

Comment (travisbrown) [reply core/src/main/scala/org/http4s/QValue.scala]: The issue is the new import pulling in `catsKernelOrderingForOrder`, which has higher precedence than the `Ordering`-from-`Ordered` stuff. You could either hide it, or only import `catsSyntaxEither`, or (what I'd do) not use `valueOr`.

Comment (rossabaker) [reply core/src/main/scala/org/http4s/QValue.scala]: Oof, thanks. I'd moved on to a different branch and was stupefied by what import was even causing it. As usual, I should have quit an hour before I did.

**Comment by ChristopherDavenport** [comment]
Test failure is a flaky test on Timeout Middleware.

**Pull request merged**